`libabsl-dev` needed on arm64 linux

**Describe the bug**
Following the build instructions, I get the following error on a fresh `arm64` Ubuntu instance on EC2:
```
[224/454] Building CXX object helio/base/CMakeFiles/base.dir/histogram.cc.o
/home/ubuntu/dragonfly/helio/base/histogram.cc: In static member function ‘static std::pair<double, double> base::Histogram::BucketLimits(unsigned int)’:
/home/ubuntu/dragonfly/helio/base/histogram.cc:41:63: note: parameter passing for argument of type ‘std::pair<double, double>’ when C++17 is enabled changed to match C++14 in GCC 10.1
   41 | inline pair<double, double> Histogram::BucketLimits(unsigned b) {
      |                                                               ^
[280/454] Building CXX object helio/util/cloud/CMakeFiles/aws_lib.dir/aws.cc.o
FAILED: helio/util/cloud/CMakeFiles/aws_lib.dir/aws.cc.o 
/usr/local/bin/ccache /usr/bin/g++ -DBOOST_ASIO_SEPARATE_COMPILATION -DBOOST_BEAST_SEPARATE_COMPILATION -DUSE_FB2 -I/home/ubuntu/dragonfly -I/home/ubuntu/dragonfly/genfiles -I/home/ubuntu/dragonfly/src -I/home/ubuntu/dragonfly/helio -Wall -Wextra -g -fPIC -fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free -fno-omit-frame-pointer -Wno-unused-parameter -march=armv8.2-a+fp16+rcpc+dotprod+crypto  -std=c++20 -DHAS_RAWMEMCHR -fdiagnostics-color=always  -O3 -DNDEBUG -MD -MT helio/util/cloud/CMakeFiles/aws_lib.dir/aws.cc.o -MF helio/util/cloud/CMakeFiles/aws_lib.dir/aws.cc.o.d -o helio/util/cloud/CMakeFiles/aws_lib.dir/aws.cc.o -c /home/ubuntu/dragonfly/helio/util/cloud/aws.cc
In file included from /home/ubuntu/dragonfly/helio/util/asio_stream_adapter.h:9,
                 from /home/ubuntu/dragonfly/helio/util/http/http_client.h:16,
                 from /home/ubuntu/dragonfly/helio/util/cloud/aws.h:12,
                 from /home/ubuntu/dragonfly/helio/util/cloud/aws.cc:5:
/home/ubuntu/dragonfly/helio/util/fiber_socket_base.h:8:10: fatal error: absl/base/attributes.h: No such file or directory
    8 | #include <absl/base/attributes.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
[285/454] Building CXX object helio/util/http/CMakeFiles/http_server_lib.dir/http_handler.cc.o
ninja: build stopped: subcommand failed.
```

**To Reproduce**
Steps to reproduce the behavior:
1. Clone the project on a `arm64` Ubuntu instance
2. Follow directions to configure and compile Dragonfly
3. Observe error

**Expected behavior**
Should not error

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Environment (please complete the following information):**
 - OS: Ubuntu 22.04
 - Kernel: `Linux ip-172-31-60-177 5.19.0-1025-aws #26~22.04.1-Ubuntu SMP Mon Apr 24 01:58:03 UTC 2023 aarch64 aarch64 aarch64 GNU/Linux`
 - Containerized?: No (Hypervisor, no container)
 - Dragonfly Version: `main`

**Reproducible Code Snippet**
N/A

**Additional context**
Adding the library `libabsl-dev` via `apt-get install -y libabsl-dev` fixed the error for me. May just need a [doc update](https://github.com/dragonflydb/dragonfly/blob/main/docs/build-from-source.md?plain=1#L13-L16).

Hm, maybe this is some other issue. I'm getting the following errors now:
```
[16/155] Building CXX object helio/util/cloud/CMakeFiles/aws_lib.dir/s3_file.cc.o
FAILED: helio/util/cloud/CMakeFiles/aws_lib.dir/s3_file.cc.o 
/usr/local/bin/ccache /usr/bin/g++ -DBOOST_ASIO_SEPARATE_COMPILATION -DBOOST_BEAST_SEPARATE_COMPILATION -DUSE_FB2 -I/home/ubuntu/dragonfly -I/home/ubuntu/dragonfly/genfiles -I/home/ubuntu/dragonfly/src -I/home/ubuntu/dragonfly/helio -Wall -Wextra -g -fPIC -fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free -fno-omit-frame-pointer -Wno-unused-parameter -march=armv8.2-a+fp16+rcpc+dotprod+crypto  -std=c++20 -DHAS_RAWMEMCHR -fdiagnostics-color=always  -O3 -DNDEBUG -MD -MT helio/util/cloud/CMakeFiles/aws_lib.dir/s3_file.cc.o -MF helio/util/cloud/CMakeFiles/aws_lib.dir/s3_file.cc.o.d -o helio/util/cloud/CMakeFiles/aws_lib.dir/s3_file.cc.o -c /home/ubuntu/dragonfly/helio/util/cloud/s3_file.cc
/home/ubuntu/dragonfly/helio/util/cloud/s3_file.cc:8:10: fatal error: libxml/xpath.h: No such file or directory
    8 | #include <libxml/xpath.h>
      |          ^~~~~~~~~~~~~~~~
compilation terminated.
[17/155] Building CXX object helio/util/cloud/CMakeFiles/aws_lib.dir/aws.cc.o
FAILED: helio/util/cloud/CMakeFiles/aws_lib.dir/aws.cc.o 
/usr/local/bin/ccache /usr/bin/g++ -DBOOST_ASIO_SEPARATE_COMPILATION -DBOOST_BEAST_SEPARATE_COMPILATION -DUSE_FB2 -I/home/ubuntu/dragonfly -I/home/ubuntu/dragonfly/genfiles -I/home/ubuntu/dragonfly/src -I/home/ubuntu/dragonfly/helio -Wall -Wextra -g -fPIC -fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free -fno-omit-frame-pointer -Wno-unused-parameter -march=armv8.2-a+fp16+rcpc+dotprod+crypto  -std=c++20 -DHAS_RAWMEMCHR -fdiagnostics-color=always  -O3 -DNDEBUG -MD -MT helio/util/cloud/CMakeFiles/aws_lib.dir/aws.cc.o -MF helio/util/cloud/CMakeFiles/aws_lib.dir/aws.cc.o.d -o helio/util/cloud/CMakeFiles/aws_lib.dir/aws.cc.o -c /home/ubuntu/dragonfly/helio/util/cloud/aws.cc
/home/ubuntu/dragonfly/helio/util/cloud/aws.cc:13:10: fatal error: rapidjson/document.h: No such file or directory
   13 | #include <rapidjson/document.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
[18/155] Building CXX object helio/util/cloud/CMakeFiles/aws_lib.dir/s3.cc.o
FAILED: helio/util/cloud/CMakeFiles/aws_lib.dir/s3.cc.o 
/usr/local/bin/ccache /usr/bin/g++ -DBOOST_ASIO_SEPARATE_COMPILATION -DBOOST_BEAST_SEPARATE_COMPILATION -DUSE_FB2 -I/home/ubuntu/dragonfly -I/home/ubuntu/dragonfly/genfiles -I/home/ubuntu/dragonfly/src -I/home/ubuntu/dragonfly/helio -Wall -Wextra -g -fPIC -fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free -fno-omit-frame-pointer -Wno-unused-parameter -march=armv8.2-a+fp16+rcpc+dotprod+crypto  -std=c++20 -DHAS_RAWMEMCHR -fdiagnostics-color=always  -O3 -DNDEBUG -MD -MT helio/util/cloud/CMakeFiles/aws_lib.dir/s3.cc.o -MF helio/util/cloud/CMakeFiles/aws_lib.dir/s3.cc.o.d -o helio/util/cloud/CMakeFiles/aws_lib.dir/s3.cc.o -c /home/ubuntu/dragonfly/helio/util/cloud/s3.cc
/home/ubuntu/dragonfly/helio/util/cloud/s3.cc:9:10: fatal error: libxml/xpath.h: No such file or directory
    9 | #include <libxml/xpath.h>
      |          ^~~~~~~~~~~~~~~~
compilation terminated.
[21/155] Performing build step for 'reflex_project'
```

@romange thank you for the hint, i have followed your suggestion but i arrived at this error instead:

```
[208/454] Building CXX object helio/base/CMakeFiles/base.dir/histogram.cc.o
/home/ubuntu/dragonfly/helio/base/histogram.cc: In static member function ‘static std::pair<double, double> base::Histogram::BucketLimits(unsigned int)’:
/home/ubuntu/dragonfly/helio/base/histogram.cc:41:63: note: parameter passing for argument of type ‘std::pair<double, double>’ when C++17 is enabled changed to match C++14 in GCC 10.1
   41 | inline pair<double, double> Histogram::BucketLimits(unsigned b) {
      |                                                               ^
[239/454] Building CXX object helio/util/cloud/CMakeFiles/aws_lib.dir/s3_file.cc.o
FAILED: helio/util/cloud/CMakeFiles/aws_lib.dir/s3_file.cc.o 
/usr/local/bin/ccache /usr/bin/g++ -DBOOST_ASIO_SEPARATE_COMPILATION -DBOOST_BEAST_SEPARATE_COMPILATION -DUSE_FB2 -I/home/ubuntu/dragonfly -I/home/ubuntu/dragonfly/genfiles -I/home/ubuntu/dragonfly/src -I/home/ubuntu/dragonfly/helio -Wall -Wextra -g -fPIC -fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free -fno-omit-frame-pointer -Wno-unused-parameter -march=armv8.2-a+fp16+rcpc+dotprod+crypto  -std=c++20 -DHAS_RAWMEMCHR -fdiagnostics-color=always  -O3 -DNDEBUG -MD -MT helio/util/cloud/CMakeFiles/aws_lib.dir/s3_file.cc.o -MF helio/util/cloud/CMakeFiles/aws_lib.dir/s3_file.cc.o.d -o helio/util/cloud/CMakeFiles/aws_lib.dir/s3_file.cc.o -c /home/ubuntu/dragonfly/helio/util/cloud/s3_file.cc
In file included from /home/ubuntu/dragonfly/helio/util/cloud/s3_file.h:7,
                 from /home/ubuntu/dragonfly/helio/util/cloud/s3_file.cc:4:
/home/ubuntu/dragonfly/helio/io/file.h:7:10: fatal error: absl/types/span.h: No such file or directory
    7 | #include <absl/types/span.h>
      |          ^~~~~~~~~~~~~~~~~~~
compilation terminated.
[240/454] Building CXX object helio/util/cloud/CMakeFiles/aws_lib.dir/s3.cc.o
FAILED: helio/util/cloud/CMakeFiles/aws_lib.dir/s3.cc.o 
/usr/local/bin/ccache /usr/bin/g++ -DBOOST_ASIO_SEPARATE_COMPILATION -DBOOST_BEAST_SEPARATE_COMPILATION -DUSE_FB2 -I/home/ubuntu/dragonfly -I/home/ubuntu/dragonfly/genfiles -I/home/ubuntu/dragonfly/src -I/home/ubuntu/dragonfly/helio -Wall -Wextra -g -fPIC -fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free -fno-omit-frame-pointer -Wno-unused-parameter -march=armv8.2-a+fp16+rcpc+dotprod+crypto  -std=c++20 -DHAS_RAWMEMCHR -fdiagnostics-color=always  -O3 -DNDEBUG -MD -MT helio/util/cloud/CMakeFiles/aws_lib.dir/s3.cc.o -MF helio/util/cloud/CMakeFiles/aws_lib.dir/s3.cc.o.d -o helio/util/cloud/CMakeFiles/aws_lib.dir/s3.cc.o -c /home/ubuntu/dragonfly/helio/util/cloud/s3.cc
In file included from /home/ubuntu/dragonfly/helio/util/cloud/s3.h:8,
                 from /home/ubuntu/dragonfly/helio/util/cloud/s3.cc:5:
/home/ubuntu/dragonfly/helio/io/io.h:7:10: fatal error: absl/types/span.h: No such file or directory
    7 | #include <absl/types/span.h>
      |          ^~~~~~~~~~~~~~~~~~~
compilation terminated.
[242/454] Building CXX object helio/util/cloud/CMakeFiles/aws_lib.dir/aws.cc.o
FAILED: helio/util/cloud/CMakeFiles/aws_lib.dir/aws.cc.o 
/usr/local/bin/ccache /usr/bin/g++ -DBOOST_ASIO_SEPARATE_COMPILATION -DBOOST_BEAST_SEPARATE_COMPILATION -DUSE_FB2 -I/home/ubuntu/dragonfly -I/home/ubuntu/dragonfly/genfiles -I/home/ubuntu/dragonfly/src -I/home/ubuntu/dragonfly/helio -Wall -Wextra -g -fPIC -fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free -fno-omit-frame-pointer -Wno-unused-parameter -march=armv8.2-a+fp16+rcpc+dotprod+crypto  -std=c++20 -DHAS_RAWMEMCHR -fdiagnostics-color=always  -O3 -DNDEBUG -MD -MT helio/util/cloud/CMakeFiles/aws_lib.dir/aws.cc.o -MF helio/util/cloud/CMakeFiles/aws_lib.dir/aws.cc.o.d -o helio/util/cloud/CMakeFiles/aws_lib.dir/aws.cc.o -c /home/ubuntu/dragonfly/helio/util/cloud/aws.cc
In file included from /home/ubuntu/dragonfly/helio/util/asio_stream_adapter.h:9,
                 from /home/ubuntu/dragonfly/helio/util/http/http_client.h:16,
                 from /home/ubuntu/dragonfly/helio/util/cloud/aws.h:12,
                 from /home/ubuntu/dragonfly/helio/util/cloud/aws.cc:5:
/home/ubuntu/dragonfly/helio/util/fiber_socket_base.h:8:10: fatal error: absl/base/attributes.h: No such file or directory
    8 | #include <absl/base/attributes.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
[244/454] Performing patch step for 'gperf_project'
ninja: build stopped: subcommand failed.
```

--- a/docs/build-from-source.md
+++ b/docs/build-from-source.md
@@ -12,14 +12,14 @@ On Debian/Ubuntu:
 
 ```bash
 sudo apt install ninja-build libunwind-dev libboost-fiber-dev libssl-dev \
-     autoconf-archive libtool cmake g++ libzstd-dev
+     autoconf-archive libtool cmake g++ libzstd-dev bison libxml2-dev
 ```
 
 On Fedora:
 
 ```bash
 sudo yum install automake boost-devel g++ git cmake libtool ninja-build libzstd-devel  \
-     openssl-devel libunwind-devel autoconf-archive patch
+     openssl-devel libunwind-devel autoconf-archive patch bison libxml2-devel
 ```
 
 ## Step 2 - clone the project

